### ISSUE TITLE
Command line autocompletion

### ISSUE THREAD

**Issue by drevell** [abcxyz/abc#24]
### TL;DR

The library that we use for parsing command line flags, https://pkg.go.dev/github.com/abcxyz/pkg/cli, now offers support for autocompletion. I don't know what's possible here, but here are some things that might perhaps be possible:

- `abc templates render ~/dir/`<tab> would autocomplete with all possible directories inside `~/dir`
- Maybe remote template URLs could autocomplete? `abc templates render github.com/abcxyz/examples`<tab> could suggest all of the available example templates in that directory
- The `--dest` flag could autocomplete as a local directory name

Originally suggested by @sethvargo.
